### PR TITLE
fix: pin unstructured-client version

### PR DESCRIPTION
**Pull request by Coniferish**
Unstructured-client added `api_key_auth` as a param to UnstructuredClient in [version 0.9.0](https://github.com/Unstructured-IO/unstructured-python-client/commit/8c93115c926765efe95c4e2a4e6bdf739fce41ad).

This pins the version of unstructured-client so users do not receive `TypeError: UnstructuredClient.__init__() got an unexpected keyword argument 'api_key_auth'`

### CODE REVIEW

File: requirements/base.in [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+python-iso639
+python-magic
 rapidfuzz
-backoff
+requests
+tabulate
 typing-extensions
-unstructured-client
+# NOTE(jennings): pinned due to later versions not supporting api_key_auth in UnstructuredClient
+unstructured-client>=0.9.0
```
Comment (cragwolfe) [inline requirements/base.in:19]: why not a higher version, like 0.15.1?

File: requirements/base.in [hunk 2]
```diff
@@ -1,19 +1,20 @@
 -c "constraints.in"
+backoff
```
Comment (cragwolfe) [inline requirements/base.in:2]: please do not make unrelated ordering changes, it obscures the diff. (if you feel strongly to reorder, it could be a standalone PR)

File: requirements/base.in (hunk 1 above)
Comment (Coniferish) [reply requirements/base.in:19]: I thought the minimum was preferred, but am happy to bump it up if that is preferred.

File: requirements/base.in (hunk 2 above)
Comment (Coniferish) [reply requirements/base.in:2]: Sounds good. I will drop that commit.

**Review by cragwolfe — APPROVED** [review]

**Comment by cragwolfe** [comment]
actually, i'm surprised nothing else changed. need to run `make pip-compile`?

**Review by cragwolfe — CHANGES_REQUESTED** [review]
needs `make pip-compile` on python3.8

File: requirements/base.in [hunk 3]
```diff
@@ -15,5 +15,6 @@ numpy
 rapidfuzz
 backoff
 typing-extensions
-unstructured-client
+# NOTE(jennings): pinned due to later versions not supporting api_key_auth in UnstructuredClient
```
Comment (qued) [inline requirements/base.in:18]: Shouldn't this be "earlier"?
```suggestion
# NOTE(jennings): pinned due to earlier versions not supporting api_key_auth in UnstructuredClient
```

**Review by qued — APPROVED** [review]
Just some minor documentation suggestions. Otherwise LGTM!

**Comment by Coniferish** [comment]
@cragwolfe or @qued 
I just resolved conflicts, but afterwards saw in the CHANGELOG that "All dependencies are now built off of the minimum version of python being 3.10". Does this mean I should re-`pip-compile` the dependencies on this branch since these updates were done in 3.8?
